lavalink/src/client.rs

```diff
@@ -376,12 +404,12 @@ impl Lavalink {
 
 #[cfg(test)]
 mod tests {
-    use super::{ClientError, Lavalink, VoiceStateHalf};
-    use static_assertions::{assert_fields, assert_impl_all};
+    use super::{ClientError, ClientErrorType, Lavalink, VoiceStateHalf};
+    use static_assertions::assert_impl_all;
     use std::{error::Error, fmt::Debug};
 
-    assert_fields!(ClientError::SendingVoiceUpdate: source);
-    assert_impl_all!(ClientError: Clone, Debug, Error, PartialEq, Send, Sync);
+    assert_impl_all!(ClientErrorType: Debug, Send, Sync);
+    assert_impl_all!(ClientError: Error, Send, Sync);
```

Comment:
Was it intentional to drop the `PartialEq` from here?

Reply:
Making the guarantee that errors are equal is probably not something that should be done (in case this error ever has a source error), so it was dropped.